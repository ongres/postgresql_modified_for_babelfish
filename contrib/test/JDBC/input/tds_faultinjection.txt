---- Inject an error in tds comm layer while reading the request
---- Test both SQL Batch and prepare exec
--SELECT inject_fault('tds_comm_throw_error', 2);
--SELECT 1;
--prepst#!# SELECT @a #!#BIGINT|-|a|-|0

---- test normal execution
--SELECT 1;

-- Inject a pre-parsing error
-- Test both SQL Batch and prepare exec
SELECT inject_fault('pre_parsing_throw_error', 2);
SELECT 1;
prepst#!# SELECT @a #!#BIGINT|-|a|-|0

-- test normal execution
SELECT 1;

-- Inject a post-parsing error
-- Test both SQL Batch and prepare exec
SELECT inject_fault('post_parsing_throw_error', 2);
SELECT 1;
prepst#!# SELECT @a #!#BIGINT|-|a|-|0

-- test normal execution
SELECT 1;

-- Inject fault to tamper TDS request
-- Test both SQL Batch and prepare exec (but prepare exec is failing now)
SELECT inject_fault('pre_parsing_tamper_request', 1);
SELECT 1;
--prepst#!# SELECT @a #!#BIGINT|-|a|-|0

