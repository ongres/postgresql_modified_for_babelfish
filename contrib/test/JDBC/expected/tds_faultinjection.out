---- Inject an error in tds comm layer while reading the request
---- Test both SQL Batch and prepare exec
--SELECT inject_fault('tds_comm_throw_error', 2);
--SELECT 1;
--prepst#!# SELECT @a #!#BIGINT|-|a|-|0

---- test normal execution
--SELECT 1;

-- Inject a pre-parsing error
-- Test both SQL Batch and prepare exec
SELECT inject_fault('pre_parsing_throw_error', 2);
~~START~~
text
enabled, pending occurrences: 2
~~END~~

SELECT 1;
~~ERROR (Code: 33557097)~~

~~ERROR (Message: error triggered from fault injection)~~

prepst#!# SELECT ? #!#BIGINT|-|a|-|0
~~ERROR (Code: 33557097)~~

~~ERROR (Message: error triggered from fault injection)~~


-- test normal execution
SELECT 1;
~~START~~
int
1
~~END~~


-- Inject a post-parsing error
-- Test both SQL Batch and prepare exec
SELECT inject_fault('post_parsing_throw_error', 2);
~~START~~
text
enabled, pending occurrences: 2
~~END~~

SELECT 1;
~~ERROR (Code: 33557097)~~

~~ERROR (Message: error triggered from fault injection)~~

prepst#!# SELECT ? #!#BIGINT|-|a|-|0
~~ERROR (Code: 33557097)~~

~~ERROR (Message: error triggered from fault injection)~~


-- test normal execution
SELECT 1;
~~START~~
int
1
~~END~~


-- Inject fault to tamper TDS request
-- Test both SQL Batch and prepare exec (but prepare exec is failing now)
SELECT inject_fault('pre_parsing_tamper_request', 1);
~~START~~
text
enabled, pending occurrences: 1
~~END~~

SELECT 1;
~~START~~
int
1
~~END~~

--prepst#!# SELECT @a #!#BIGINT|-|a|-|0

